### DISCUSSION TITLE
scfinder configs for running in real-time

### DISCUSSION THREAD

**Discussion by dufangqing** [SED-EEW/SED-EEW-SeisComP-contributions#54] [Q&A]
Hi there,

To run scfinder in realtime, I simply executed "seiscomp-finder exec scfinder --debug', because I assume scfinder would read inventory and record streams from the host by default. 

However, in my case, it seems scfinder was only able to read the inventory from the host but not the real-time streaming waveforms since I kept getting the "too few stations in raw data list, exit trigger check!" info. 

I wonder if there are extra configs that I have to specify and do you think the connection error message I get from the log is critical for the issue?

> ...
> 16:01:29 [info] load_templates: Time passed in seconds for loading templates: 1.13259
> 16:01:29 [debug] FinDer version is finder-3.2.4_2023-10-31
> 16:01:29 [debug] GMT version is: 5.4.5
> 16:01:29 [info] Throttle Finder processing to an interval of 1.000000s
> 16:01:29 [info] Starting record acquisition
> 16:01:29 [info] Starting message thread
**16:01:29 [error] SocketException: Socket connect error; Try to reconnect**
> 16:01:30 [info] determine_trigger: too few stations in raw data list, exit trigger check!
> 16:01:31 [info] determine_trigger: too few stations in raw data list, exit trigger check!
> 16:01:33 [info] determine_trigger: too few stations in raw data list, exit trigger check!
> 16:01:35 [info] determine_trigger: too few stations in raw data list, exit trigger check!
> ....

ps., a little configuration issue that I noticed: if adding "database.inventory = host.docker.internal/seiscomp" to the config file, I would get the error below, one may just exclude this specification to avoid the issue... 

> 16:01:08 [info] Loading complete inventory from host.docker.internal/seiscomp
> 16:01:08 [error] host.docker.internal/seiscomp not found

Thanks a lot,
Du

**Comment by FMassin** [comment]
Dear @dufangqing 

#### Socket connet error
```bash
16:01:29 [error] SocketException: Socket connect error; Try to reconnect
```
Means that `scfinder` cannot connect to the `seedlink` server. This message is common to all seiscomp modules using seismic data (i.e., `scautopick`, `scamp`, and `sceewenv`  & `scfinder` from the EEW addons). So I think you need to go back to `slinktool` and `seedlink` to understand what's going on. I suspect `seedlink` to be crashed.

#### database.inventory error
Depending on the version, the global seiscomp  parameter `database.inventory` might require database type (e.g,. `mysql://`), your response is great, you can also fix it with:
```bash
database.inventory = mysql://host.docker.internal/seiscomp
```
But this kind of complex config is irrelevant to 99% of users so I will remove it from the documentation. Thanks a lot for spotting this.

**Reply by dufangqing** [reply]
Thank you very much @FMassin. The connection problem was solved by specifying `--record-url slink://host.docker.internal:18000`.

**Reply by dufangqing** [reply]
however, I received data latency warnings (e.g., `09:05:44 [warning] OX.CSM..HNN: max delay exceeded: 17.408047s`) while running scfinder in real-time. But I guess this latency is from my seismic network itself?

**Reply by FMassin** [reply]
Yes, it should reflect what's called _latency_ provided by `slinktool -p -S 'OX.CSM:HNN' host.docker.internal:18000`.

**Reply by dufangqing** [reply]
Thanks! Indeed there's latency in the network.
